geoutils/raster/raster.py

```diff
@@ -2084,10 +2085,70 @@ def reproject(
         dtype: DTypeLike | None = None,
         resampling: Resampling | str = Resampling.bilinear,
         force_source_nodata: int | float | None = None,
+        *,
+        inplace: Literal[False] = False,
         silent: bool = False,
         n_threads: int = 0,
         memory_limit: int = 64,
     ) -> RasterType:
+        ...
+
+    @overload
+    def reproject(
+        self: RasterType,
+        ref: RasterType | str | None = None,
+        crs: CRS | str | int | None = None,
+        res: float | abc.Iterable[float] | None = None,
+        grid_size: tuple[int, int] | None = None,
+        bounds: dict[str, float] | rio.coords.BoundingBox | None = None,
+        nodata: int | float | None = None,
+        dtype: DTypeLike | None = None,
+        resampling: Resampling | str = Resampling.bilinear,
+        force_source_nodata: int | float | None = None,
+        *,
+        inplace: Literal[True],
+        silent: bool = False,
+        n_threads: int = 0,
+        memory_limit: int = 64,
+    ) -> None:
+        ...
+
+    @overload
+    def reproject(
+        self: RasterType,
+        ref: RasterType | str | None = None,
+        crs: CRS | str | int | None = None,
+        res: float | abc.Iterable[float] | None = None,
+        grid_size: tuple[int, int] | None = None,
+        bounds: dict[str, float] | rio.coords.BoundingBox | None = None,
+        nodata: int | float | None = None,
+        dtype: DTypeLike | None = None,
+        resampling: Resampling | str = Resampling.bilinear,
+        force_source_nodata: int | float | None = None,
+        *,
+        inplace: bool = False,
+        silent: bool = False,
+        n_threads: int = 0,
+        memory_limit: int = 64,
+    ) -> RasterType | None:
+        ...
+
+    def reproject(
```

Comment:
Why is there a need for a 4th overloading? The last two are identical apart from the star...

Reply:
So it's 3 overloading + the normal function def

Reply:
Those overloading are quite annoying, both to write and in term of space use... I hope they can find an easier syntax later.
Soon they will make up half of the code lines 😉